go/tools/builders/compilepkg.go

```diff
@@ -270,16 +271,17 @@ func compileArchive(
 	// constraints.
 	compilingWithCgo := haveCgo && cgoEnabled
 
+	keepForNogo := func(s string) bool { return !ignoreForNogo[s] }
```

Comment:
Nit: i think you could inline the whole filterSlice func here instead of declaring a utility func + a filter func.

Reply:
Done